wyzesense2mqtt.py

```diff
@@ -1,229 +1,392 @@
+    SENSORS[sensor_mac]['name'] = f"Wyze Sense {sensor_mac}"
+    SENSORS[sensor_mac]['class'] = (
+            "motion" if (sensor_type == "motion")
+            else "opening"
+    )
+    if (sensor_version is not None):
+        SENSORS[sensor_mac]['sw_version'] = sensor_version
+
+    LOGGER.info("Writing Sensors Config File")
+    write_yaml_file(SENSORS_CONFIG_FILE, SENSORS)
+
+
+# Send discovery topics
+def send_discovery_topics(sensor_mac):
+    global SENSORS, CONFIG
+    LOGGER.info(f"Publishing discovery topics for {sensor_mac}")
+
+    sensor_name = SENSORS[sensor_mac]['name']
+    sensor_class = SENSORS[sensor_mac]['class']
+    if (SENSORS[sensor_mac].get('sw_version') is not None):
+        sensor_version = SENSORS[sensor_mac]['sw_version']
+    else:
+        sensor_version = ""
 
     device_payload = {
-        'identifiers': ["wyzesense_{0}".format(sensor_mac), sensor_mac],
+        'identifiers': [f"wyzesense_{sensor_mac}", sensor_mac],
         'manufacturer': "Wyze",
-        'model': ("Motion Sensor" if sensor_type == "motion" else "Contact Sensor"),
-        'name': ("Wyze Sense Motion Sensor" if sensor_type == "motion" else "Wyze Sense Contact Sensor")
+        'model': (
+                "Sense Motion Sensor" if (sensor_class == "motion")
+                else "Sense Contact Sensor"
+        ),
+        'name': sensor_name,
+        'sw_version': sensor_version
     }
 
-    device_classes = {
+    entity_payloads = {
         'state': {
-            'name': "Wyze Sense {0}".format(sensor_mac),
-            'dev_cla': ("motion" if sensor_type == "motion" else "opening"),
+            'name': sensor_name,
+            'dev_cla': sensor_class,
             'pl_on': "1",
             'pl_off': "0",
-            'json_attr_t': WYZESENSE2MQTT_TOPIC_ROOT + sensor_mac
+            'json_attr_t': f"{CONFIG['self_topic_root']}/{sensor_mac}"
         },
         'signal_strength': {
-            'name': "Wyze Sense {0} Signal Strength".format(sensor_mac),
+            'name': f"{sensor_name} Signal Strength",
             'dev_cla': "signal_strength",
             'unit_of_meas': "dBm"
         },
         'battery': {
-            'name': "Wyze Sense {0} Battery".format(sensor_mac),
+            'name': f"{sensor_name} Battery",
             'dev_cla': "battery",
             'unit_of_meas': "%"
         }
     }
 
-    # Send Discovery Topics
-    for device_class in device_classes :
-        device_classes[device_class]['val_tpl'] = "{{{{ value_json.{0} }}}}".format(device_class)
-        device_classes[device_class]['uniq_id'] = "wyzesense_{0}_{1}".format(sensor_mac, device_class)
-        device_classes[device_class]['stat_t'] = WYZESENSE2MQTT_TOPIC_ROOT + sensor_mac
-        device_classes[device_class]['dev'] = device_payload
-        sensor_type = ("binary_sensor" if device_class == "state" else "sensor")
+    for entity in entity_payloads:
+        entity_payloads[entity]['val_tpl'] = f"{{{{ value_json.{entity} }}}}"
```

Comment:
I am curious if it is necessary to access like this since as far as I know entity == entity_payloads[entity]

More just a readability/cleanliness thing than anything else.

Reply:
I believe I originally tried doing those as entity['val_tpl'] (etc.), but for some reason Python was not actually copying the full dict into the entity variable, but only the key. It's possible I was doing something wrong at the time though, as I would find it more readable to not have to dive back into the full dict. I'll see if I can play some with this and see what happens.

Reply:
Found a way to do it. How's this look in comparison:
`
for entity, entity_dict in entity_payload.items() :
    entity_dict['val_tpl'] = "{{ value_json.{0} }}".format(entity)
    entity_dict['uniq_id'] = "wyzesense_{0}_{1}".format(sensor_mac, entity)
    entity_dict['stat_t'] = WYZESENSE2MQTT_TOPIC_ROOT + sensor_mac
    entity_dict['dev_cla'] = entity
    entity_dict['dev'] = device_payload
    dev_type = entity_dict['type']
    del entity_dict['type']
`

Reply:
Decided it was more readable and pushed the changes. Resolving.